src/bionics.cpp

```diff
@@ -1492,14 +1492,15 @@ bool player::install_bionics(const itype &type, int skill_level)
         add_msg(m_good, _("Successfully installed %s."), bionics[bioid].name.c_str());
         add_bionic(bioid);
 
-        if (bioid == "bio_ears") {
+        if( bioid == "bio_eye_optic" && g->u.has_trait( "HYPEROPIC" ) ) {
+            g->u.remove_mutation( "HYPEROPIC" );
```

Comment:
And while you're changing this, remove the `g->u.` as well. This could should check and modify the implicit `this` object, be it the player character or whatever else.